scripts/fetch_binary.d

```diff
@@ -22,80 +24,100 @@ string suffix(Platform p) => p.startsWith("windows") ? "7z" : "tar.xz";
     Compiler.dmd: CompilerInfo(
-        (platform, compilerVersion) => "http://downloads.dlang.org/releases/2.x/%s/dmd.%s.%s.%s"
-            .format(compilerVersion, compilerVersion, platform, suffix(platform)),
-        [
-            "linux", "osx", "freebsd-64", "windows"
-        ],
+        (platform, compilerVersion) =>
+            "http://downloads.dlang.org/releases/2.x/%s/dmd.%s.%s.%s"
+                .format(compilerVersion, compilerVersion, platform, suffix(platform)),
+            [
+                "linux", "osx", "freebsd-64", "windows"
+            ],
     ),
     Compiler.ldc: CompilerInfo(
-        (platform, compilerVersion) => "https://github.com/ldc-developers/ldc/releases/download/v%s/ldc2-%s-%s.%s"
-            .format(compilerVersion, compilerVersion, platform, suffix(platform)),
-        [
-            "android-aarch64", "android-armv7a",
-            "freebsd-x86_64",
-            "linux-aarch64", "linux-x86_64",
-            "osx-arm64", "osx-x86_64",
-            "windows-x64", "windows-x86"
-        ],
+        (platform, compilerVersion) =>
+            "https://github.com/ldc-developers/ldc/releases/download/v%s/ldc2-%s-%s.%s"
+                .format(compilerVersion, compilerVersion, platform, suffix(platform)),
+            [
+                "android-aarch64", "android-armv7a",
+                "freebsd-x86_64",
+                "linux-aarch64", "linux-x86_64",
+                "osx-arm64", "osx-x86_64",
+                "windows-x64", "windows-x86"
+            ],
     ),
 ];
 
 void main(string[] args) {
-    Version compilerVersion;
+    Version[] compilerVersions;
     Compiler compiler = Compiler.ldc;
     bool dryRun = true;
 
-    auto help = args.getopt(
-        "version", &compilerVersion,
-        "compiler", &compiler,
-        "dry-run", &dryRun,
-    );
+    auto parseCLI(string[] args) {
+        std.getopt.arraySep = ",";
+        return args.getopt(
+            "versions", &compilerVersions,
+            "compiler", &compiler,
+            "dry-run", &dryRun,
+        );
+    }
 
-    compilerVersion = compilerVersion
-        ? compilerVersion
-        : compiler == Compiler.ldc
-        ? "1.34.0"
-        : "2.105.0";
+    bool helpWanted = false;
+    try {
+        auto cli = parseCLI(args);
+        helpWanted = cli.helpWanted;
+    } catch (GetOptException) {
+        helpWanted = true;
+    }
 
-    if (help.helpWanted)
-        defaultGetoptPrinter("Some information about the program.", help.options);
+    if (helpWanted) {
+        auto w = stderr.lockingTextWriter();
+        defaultGetoptFormatter(
+            w,
+            "fetch_binary.d - " ~
+                "tool for fetching binary releases of DMD and LDC",
+            parseCLI(args[0 .. 1]).options,
+        );
+        return;
+    }
+
+    compilerVersions = compilerVersions.length
+        ? compilerVersions
+        : compiler == Compiler.ldc
+        ? [ "1.34.0" ]
+        : [ "2.105.0" ];
```

Comment:
For avoiding maintenance burden, it's better to either omit this default altogether, or use the newest version in `supported-binary-versions.json` as the default.

Reply:
I have a big rewrite of this tool pending, and I'll address your concern in the next PR.